### PR TITLE
chore(query): make user setting depend on a lightweight UserSettingValue

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

make user setting depend on a lightweight UserSettingValue  instead of DataValue.

It's meta compatible with before.

### CODE REVIEW

**Review by Xuanwo — APPROVED** [review]

File: src/meta/types/src/user_setting.rs [hunk 1]
```diff
@@ -12,33 +12,80 @@
...
-impl UserSetting {
-    pub fn create(name: &str, value: DataValue) -> UserSetting {
-        UserSetting {
-            name: name.to_string(),
-            value,
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq)]
+pub enum UserSettingValue {
+    UInt64(u64),
+    String(Vec<u8>),
```
[the remark this answers is not in the document]
Comment (sundy-li) [reply src/meta/types/src/user_setting.rs:33]: Yes, I'm working on it.

**Comment by sundy-li** [comment]
```
❯ mysql --port 3307 --host 127.0.0.1 --user default --prompt=':) '
ERROR 1815 (HY000): invalid type: string "MySQL", expected a sequence at line 1 column 47

```

`String` is not compatible with `Vec<u8> `

**Comment by BohuTANG** [comment]
Will this pr cause backward compatibility?
If we already serialized `DataValue` to metasrv and deserialized by the new `UserSettingValue`.

**Comment by drmingdrmer** [comment]
> Will this pr cause backward compatibility? If we already serialized `DataValue` to metasrv and deserialized by the new `UserSettingValue`.

No. `DataValue::String(Vec<u8>)` has been changed to `UserSettingValue::String(String)`.
It has to use identical types or create another field with a different type.

**Comment by drmingdrmer** [comment]
The tests of compatibility between old and new query are defined in these 2 files:

`tests/fuse-compat/compat-logictest/fuse_compat_write` writes data with an old query.
`tests/fuse-compat/compat-logictest/fuse_compat_read` reads the data with a new query.

`fuse-compat` only guarantees backward compatibility. It does not test whether an old query can read data that is written with a new query.

**Review by drmingdrmer — APPROVED** [review]
Great

**Review by BohuTANG — APPROVED** [review]

**Pull request merged**